### PR TITLE
feat: track factory on its

### PR DESCRIPTION
**Pull request by Foivos**

### CODE REVIEW

File: contracts/InterchainTokenService.sol [hunk 1]
```diff
@@ -20,6 +20,7 @@ import { ITokenManagerDeployer } from './interfaces/ITokenManagerDeployer.sol';
 import { IInterchainTokenDeployer } from './interfaces/IInterchainTokenDeployer.sol';
 import { IInterchainTokenExecutable } from './interfaces/IInterchainTokenExecutable.sol';
 import { IInterchainTokenExpressExecutable } from './interfaces/IInterchainTokenExpressExecutable.sol';
+import { IInterchainTokenFactory } from './interfaces/IInterchainTokenFactory.sol';
```
Comment (milapsheth) [inline contracts/InterchainTokenService.sol:23]: unused

File: contracts/InterchainTokenService.sol [hunk 2]
```diff
@@ -48,6 +49,7 @@ contract InterchainTokenService is
 
     IAxelarGateway public immutable gateway;
     IAxelarGasService public immutable gasService;
+    address public immutable tokenFactory;
```
Comment (milapsheth) [inline contracts/InterchainTokenService.sol:52]:
```suggestion
    address public immutable interchainTokenFactory;
```
for consistency

File: contracts/InterchainTokenService.sol [hunk 3]
```diff
@@ -272,6 +277,9 @@ contract InterchainTokenService is
         uint256 gasValue
     ) external payable whenNotPaused returns (bytes32 tokenId) {
         address deployer = msg.sender;
+        
+        if(deployer == tokenFactory) deployer = address(0);
```
Comment (milapsheth) [inline contracts/InterchainTokenService.sol:281]:
```suggestion
        if (deployer == tokenFactory) deployer = address(0);
```

File: contracts/InterchainTokenService.sol [hunk 4]
```diff
@@ -304,8 +312,12 @@ contract InterchainTokenService is
         uint8 decimals,
         bytes memory distributor,
         uint256 gasValue
-    ) external payable whenNotPaused {
-        bytes32 tokenId = interchainTokenId(msg.sender, salt);
+    ) external payable whenNotPaused {        
+        address deployer = msg.sender;
+
+        if(deployer == tokenFactory) deployer = address(0);
```
Comment (milapsheth) [inline contracts/InterchainTokenService.sol:318]:
```suggestion
        if (deployer == tokenFactory) deployer = address(0);
```

File: scripts/deploy.js [hunk 5]
```diff
@@ -70,7 +83,13 @@ async function deployTokenManagerImplementations(wallet, interchainTokenServiceA
     return implementations;
 }
 
-async function deployAll(wallet, chainName, evmChains = [], deploymentKey = 'interchainTokenService') {
+async function deployAll(
+    wallet,
+    chainName,
+    evmChains = [],
+    deploymentKey = 'interchainTokenService',
```
Comment (milapsheth) [inline scripts/deploy.js:90]:
```suggestion
    deploymentKey = 'InterchainTokenService',
```

File: contracts/InterchainTokenFactory.sol [hunk 6]
```diff
@@ -53,11 +53,11 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
     }
 
     function interchainTokenId(address deployer, bytes32 salt) public view returns (bytes32 tokenId) {
-        tokenId = service.interchainTokenId(address(this), interchainTokenSalt(chainNameHash, deployer, salt));
+        tokenId = service.interchainTokenId(address(0), interchainTokenSalt(chainNameHash, deployer, salt));
```
Comment (milapsheth) [inline contracts/InterchainTokenFactory.sol:56]: Create a constant for `address(0)`, maybe `TOKEN_FACTORY_DEPLOYER` and add a comment on it?

**Review by milapsheth — APPROVED** [review]

File: contracts/InterchainTokenService.sol (hunk 2 above)
Comment (Foivos) [reply contracts/InterchainTokenService.sol:51]: changed

File: contracts/InterchainTokenService.sol (hunk 3 above)
Comment (Foivos) [reply contracts/InterchainTokenService.sol]: fixed

File: contracts/InterchainTokenService.sol (hunk 4 above)
Comment (Foivos) [reply contracts/InterchainTokenService.sol]: fixed

**Pull request merged**